### PR TITLE
Adding `trand` and `randh` tests, and style-guide modifications. 

### PR DESCRIPTION
**Pull request by narner**

### CODE REVIEW

File: util/style_guide.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ to these guidelines.
 # C Code Conventions
 
 The C code in Soundpipe follows most of the [C style guidelines](http://www.suckless.org/style) found on the
```
Comment (PaulBatchelor) [inline util/style_guide.md]: This is a broken link. 

Replace with: http://suckless.org/coding_style

File: util/style_guide.md [hunk 2]
```diff
@@ -39,13 +39,14 @@ If you are using DSP code or algorithms from someone else, keep it in the header
 There are several ported Csound opcodes you can use as reference (metro is a good example).
 
 
+
 # Naming
 
 Choose good names for your variables so your code is clear. In C, a good variable name
-has the right balance of terseness and verboseness for maximum clarity and grokking. 
+has the right balance of terseness and verboseness for maximum clarity and grokking. As an example, abbreviating "frquency" to "freq" is acceptable, but abbreviating "filter cutoff" to "fc" is not.
```
Comment (PaulBatchelor) [inline util/style_guide.md]: "frquency" to "frequency"

File: util/style_guide.md [hunk 3]
```diff
@@ -68,3 +69,10 @@ If you are porting Csound code, there is often a tendency to put a, k, or i in f
 of a variable to indicate audio, control, or init-rate variables. Ditch the a's and k's...
 i's can be allowed.
 
+# Tests
+
+If you are adding a test for a module you wrote, make sure the test passes before making a pull-request. 
+
+#Algorithm Explanations
```
Comment (PaulBatchelor) [inline util/style_guide.md]: Make "#Algorithm Explanations" "# Algorithm Explanations"

File: util/style_guide.md [hunk 4]
```diff
@@ -68,3 +69,10 @@ If you are porting Csound code, there is often a tendency to put a, k, or i in f
...
+
+DSP code, especially when written in C, can be difficult to read. Algorithm explanations should be kept in modules/data, and not in the source code itself. Make sure any explanations are clear and concise.
```
Comment (PaulBatchelor) [inline util/style_guide.md]: Break this line up.

There should also be a note in here in general about line width being approx 80 characters/line. This is ideal, though we break the rule quite a bit.

**Comment by narner** [comment]
@PaulBatchelor Just pushed up the fixes.

**Comment by narner** [comment]
Ah hold on missed one.

File: test/t/t_trand.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+} UserData;
+
+int t_trand(sp_test *tst, sp_data *sp, const char *hash) 
+{
+    uint32_t n;
+    int fail = 0;
+    SPFLOAT freq;
+    
+    UserData ud;
+    sp_create(&sp);
```
Comment (PaulBatchelor) [inline test/t/t_trand.c]: remove this

File: test/t/t_trand.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    sp->len = 44100 * 5;  
+
+    for(n = 0; n < tst->size; n++) {
+        SPFLOAT osc = 0, trand = 0, met = 0;
+        sp_metro_compute(sp, ud.met, NULL, &met);
+        sp_trand_compute(sp, ud.trand, &met, &trand);
+        ud.osc->freq = trand;
+        sp_osc_compute(sp, ud.osc, NULL, &osc);
+        sp_test_add_sample(tst, sp->out[0]);
```
Comment (PaulBatchelor) [inline test/t/t_trand.c]: sp->out[0] isn't actually being set. What we'd get here is undefined behavior, which is yucky.

The value "osc" is the value we want to add to the test buffer. You can do two things:
Set sp->out[0] before adding the sample:
sp->out[0] = osc;

or we can just do a way with sp->out[0] and just add osc to the test buffer directly:

sp_test_add_sample(tst, osc)

**Pull request merged**